### PR TITLE
feat!: Actionable Error

### PR DESCRIPTION
**Pull request by jainsahab**
Adding the following public methods in `DatastoreException` to provide more info related to errors.
-getReason()
-getDomain()
-getMetaData
-getErrorDetails()

depends on #1240

### CODE REVIEW

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/DatastoreException.java [hunk 1]
```diff
@@ -38,22 +43,85 @@ public final class DatastoreException extends BaseHttpServiceException {
...
+   * @return An object containing getters for structured objects from error_details.proto.
+   */
+  public ErrorDetails getErrorDetails() {
+    if (this.apiException != null) {
+      return this.apiException.getErrorDetails();
+    }
+    return null;
+  }
+
+  @Override
```
Comment (kolea2) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/DatastoreException.java:122]: missing javadoc

**Review by kolea2 — COMMENTED** [review]
nit - let's rename the title of this PR to `feat!:...` so it will explicitly show up as a breaking change in release notes

File: google-cloud-datastore/clirr-ignored-differences.xml [hunk 2]
```diff
@@ -26,4 +26,14 @@
         <method>com.google.cloud.datastore.AggregationResults runAggregation(com.google.cloud.datastore.AggregationQuery)</method>
         <differenceType>7012</differenceType>
     </difference>
+    <difference>
```
Comment (kolea2) [inline google-cloud-datastore/clirr-ignored-differences.xml:29]: please add a comment explaining what change was made

Comment (jainsahab) [reply google-cloud-datastore/clirr-ignored-differences.xml:29]: Done

**Review by kolea2 — APPROVED** [review]

**Pull request merged**